### ISSUE TITLE
Long trips spanning multiple days

### ISSUE THREAD

**Issue by ischmidt20** [Esri/public-transit-tools#147]
I realize this probably won't be noticed as this tool has been deprecated, but I don't have ArcGIS Pro so I'm stuck using this tool in ArcCatalog and ArcMap.

How can I get this tool to deal with trips that might span multiple days? For example, I'm keen on building a network with major US/Canada inter-city rail lines, using GTFS data from both VIARail and Amtrak. When I run the tool on the VIARail data, I get the following error:

`Column "arrival_time in file C:\.....\viarail\stop_times.txt has an invalid value:101:01:00`

Upon further inspection, it appears that this time corresponds to the transcontinental route, where times with hours greater than 24 signal the following the day. It appears this train leaves Toronto at 9:45 AM, then pulls into this stop at 5:01 AM on the 5th day of the trip.

How can I get around this issue? Should I manipulate the GTFS files to essentially start a new trip on each day? I'm just getting myself familiar with GTFS, but it should be possible to get it where I have an equivalent schedule, but without such large time values.

Any other suggestions?

EDIT: Just realized I didn't drop the link to the VIA Rail GTFS: https://www.viarail.ca/en/developer-resources

**Comment by mmorang** [comment]
Your question has been noticed and understood!

I just tested your data with the new tools in ArcGIS Pro, and it works correctly there.  Because it's working there, I will not be able to make any updates to these old, deprecated tools to support this case.

If your organization is up to date on its ArcGIS licensing and maintenance, you _should_ be able to get a copy of ArcGIS Pro.  You would need to talk to your Esri representative to find out what to do.  If you don't know who that is or need more help, respond here, and I will help you figure it out.

**Comment by ischmidt20** [comment]
Thanks for the quick reply! I'm actually a university student, and I obtained a year-long ArcGIS license through a GIS class I took last semester. We received EVA codes by e-mail back in August, but since I'm not in the class anymore, I don't know how feasible getting an ArcGIS Pro license would be. In any case, I should be able to re-work the GTFS to get around the issue.

On a completely unrelated note, how can we add locations and get them to snap to the network without including streets? After removing those extreme times, I followed your instructions here: https://github.com/Esri/public-transit-tools/blob/master/add-GTFS-to-a-network-dataset/UsersGuide.md, but I ignored everything to do with connectivity because I don't wish to use streets. Whenever I try to calculate a route or generate a service area using TransitTime, my locations are always in a non-traversable location, even when I load exactly the stops themselves. I remember when I built the network I got a bunch of `SourceName: Stops, ObjectID: , Standalone user-defined junction is detected.` I also notice that there are no junctions in the network. That's probably the cause of the issue, but what is the solution?

I understand that using these tools without streets kind of makes them less "realistic," but with a national dataset, it would be a lot easier to just model between the stations themselves.

Another note—a suggestion—it would be a cool addition if you incorporated the shapes.txt file (if it exists) when drawing the transit lines. It would give some meaning to length, and would just look better for display purposes. Thanks again for your help!

**Comment by mmorang** [comment]
Many universities have a university-wide ArcGIS license, so you might ask at the library or computer lab whether ArcGIS Pro is available somewhere.  You might not be able to install it on your own machine, but I'd be someone has it somewhere.  I realize this isn't the most convenient time to have to go to a computer lab, though.  Not sure if your area is as quarantined as mine is right now.

If the network has no junctions and you can't locate on anything, my guess is that it did not build successfully.  If you had enough of the right kind of build errors, it probably just isn't functioning at all.  Note that if you aren't using any street  features,  then all the  junctions are represented by transit stops, so there won't be any system junctions.  So, that might not actually be a problem.

Technically you shouldn't need the streets, and locating on transit stops should work.  I think the edits you made to the transit schedules are the more likely culprit.  The transit lines are generated by drawing a straight line between stops that are directly connected.  That information is determined by looking at the schedules.  If you just removed a bunch of rows from stop_times.txt, then there might be a bunch of stops that never have any trips that visit them, and those would show up as standalone user-defined junctions.

Anyway, I can't really tell what's wrong, but unfortunately I also can't really spend a ton of time debugging this now-deprecated tool anymore.  I wish you the best of luck!

**Comment by ischmidt20** [comment]
Yeah, there are few computers in a couple libraries that have ArcGIS installed, but it's more convenient to work on my personal computer, especially now when campus is basically closed. Hope you are staying safe!

I only deleted a few rows from a couple trips, and the locations I was testing with weren't on those trips. I still saw all the edge lines generate as usual. Plus, creating a route with "length" as the impedance tool works just fine.

In any case, I tried the [Amtrak](https://transit.land/feed-registry/operators/o-9-amtrak) gtfs file, without any modifications, which built without any errors (yay!) However, when I load the network into ArcMap, I see the edges as expected, but the junctions set is empty again. I still see a `Warning: Location ... in "Stops" is on a non-traversable network element position` for every location whenever I try to use TravelTime_WithTransit as impedance. It's weird, as I can still draw a route with plain length as the impedance, so clearly the junctions are recognized somehow. I followed the suggestions under "My analysis never uses the transit lines. It only uses the streets." in your troubleshooting guide, but no luck.

Thanks for your help—I'll keep updating this with my progress just in case anyone else runs into similar issues in the future.

**Comment by mmorang** [comment]
Hello.  I see you edited your post a few times since I initially read it.  Were you able to get your project done?

**Comment by ischmidt20** [comment]
Yes, I was able to get it working, but only after adding another shapefile representing the tracks themselves (which I treated as streets), so the stop points wouldn't be clearly coincident with the lines. It appears that a requirement of the tool is that there be a streets layer that the stop points must snap to, and that stop points must not lie directly on the streets. So using the straight transit lines themselves as "streets" is an attempted workaround that doesn't work.
